opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1328,11 +1466,37 @@ namespace detail {
         const int np = wells().number_of_phases;
         const int nw = wells().number_of_wells;
 
-        V bhp_targets  = V::Zero(nw);
-        V rate_targets = V::Zero(nw);
-        M rate_distr(nw, np*nw);
+        const ADB& aqua   = subset(state.qs, Span(nw, 1, BlackoilPhases::Aqua*nw));
```

Comment:
Rewrote for three phase correctness. I think it is correct, but please take a quick look if it looks sensible.